Applications/Spire/Source/Ui/DropDownList.cpp

```diff
@@ -0,0 +1,255 @@
+#include "Spire/Ui/DropDownList.hpp"
+#include <QFocusEvent>
+#include <QKeyEvent>
+#include <QScrollBar>
+#include <QVBoxLayout>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/ScrollArea.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+namespace {
+  const auto BORDER_PADDING = 2;
+}
+
+DropDownList::DropDownList(std::vector<DropDownItem*> items,
+    bool is_click_activated, QWidget* parent)
+    : DropDownWindow(is_click_activated, parent),
+      m_max_displayed_items(5) {
+  m_scroll_area = new ScrollArea(this);
+  m_scroll_area->setFocusProxy(parent);
+  m_scroll_area->setWidgetResizable(true);
+  auto main_widget = new QWidget(this);
+  m_layout = new QVBoxLayout(main_widget);
+  m_layout->setContentsMargins({});
+  m_layout->setSpacing(0);
+  m_scroll_area->setWidget(main_widget);
+  initialize_widget(m_scroll_area);
+  set_items(items);
+  parent->installEventFilter(this);
+}
+
+
+
+bool DropDownList::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == parent()) {
+    if(event->type() == QEvent::KeyPress) {
+      auto e = static_cast<QKeyEvent*>(event);
+      switch(e->key()) {
+        case Qt::Key_Down:
+          if(isVisible()) {
+            activate_next();
+            return true;
+          }
+          break;
+        case Qt::Key_Up:
+          if(isVisible()) {
+            activate_previous();
+            return true;
+          }
+          break;
+        case Qt::Key_Enter:
+        case Qt::Key_Return:
+          if(isVisible() && m_highlight_index) {
+            on_item_selected(get_widget(*m_highlight_index)->get_value(),
+              *m_highlight_index);
+            return true;
+          }
+          break;
+      }
+    }
+  }
+  return DropDownWindow::eventFilter(watched, event);
+}
+
+void DropDownList::hideEvent(QHideEvent* event) {
+  m_scroll_area->verticalScrollBar()->setValue(0);
```

Comment:
Why is the value set to 0 and why are all the items removed?

Reply:
I'm not seeing where the list items get removed, but the old default was to have the list scroll back to the top and remove the current highlight. I'm not sure if the filtered drop down menu will use this, but I'll remove it.

Reply:
Okay thanks for clarifying.